tests/integration_tests/test_login.py

```diff
@@ -213,3 +216,83 @@ async def test_server_double_login(lobby_server):
         'style': 'error',
         'text': 'You have been signed out because you signed in elsewhere.'
     }
+
+
+async def test_server_valid_login_with_token(lobby_server, api_priv_key):
+    proto = await connect_client(lobby_server)
+    await proto.send_message({
+        'command': 'auth',
+        'version': '1.0.0-dev',
+        'user_agent': 'faf-client',
+        'token': jwt.encode({
+          "user_id": 1,
+          "user_name": "test",
+          "scope": [],
+          "exp": int(time() + 10),
+          "authorities": [],
+          "non_locked": True,
+          "jti": "",
+          "client_id": ""
+        }, api_priv_key, algorithm='RS256').decode(),
+        'unique_id': 'some_id'
+    })
+
+    msg = await proto.read_message()
+    assert msg['command'] == 'irc_password'
+    msg = await proto.read_message()
+    assert msg == {
+        'command': 'welcome',
+        'me': {
+            'clan': '678',
+            'country': '',
+            'global_rating': [2000.0, 125.0],
+            'id': 1,
+            'ladder_rating': [2000.0, 125.0],
+            'login': 'test',
+            'number_of_games': 5
```

Comment:
Well the numbers come from the test data so I think it's fine. I like checking the entire message when possible because it means if anything changes at all you will get a failing test. Like if suddenly the global rating started being sent as `1500, 500` then that probably means some code is not loading from the database correctly anymore.